### PR TITLE
Use DocsPage Component

### PR DESCRIPTION
**Pull request by jonathantneal**
- 🔍 [Preview - Netlify](https://jn-use-page-component--nomad-website.netlify.com/)
- 🎟 [Issue - Asana](https://app.asana.com/0/1100423001970639/1166026584199635/f)
- 📘 [Docs - GitHub](https://github.com/hashicorp/web-components/blob/jn.add-docs-page-component/packages/docs-page/readme.md)

This Pull Request replaces the internal `<DocsPage />` component with the Web Components version.

#### Problem

We do not have a unified docs page between Nomad and Vault, and instead the code is copy-pasted between them at the moment. This is not an ideal pattern.

#### Solution

We will move the `<DocsPage />` component into the Web Component library for easier sharing between docs sites.

#### Related

This component includes a Nomad override, which is preserved by request.

- 🎟 [Related Issue - More idiomatic implementation of "jump to section" feature - Asana](https://app.asana.com/0/1100423001970639/1160656182754009)

### CODE REVIEW

File: website/layouts/api.jsx [hunk 1]
```diff
@@ -1,21 +1,42 @@
...
     )
   }
+  /*
+    data,
+  order,
+  currentPage,
+  category,
+  Link,
+  product
+  */
```
Comment (jescalan) [inline website/layouts/api.jsx]: 🔪

File: website/layouts/api.jsx [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
 
-  ApiLayout.getInitialProps = getInitialProps
+  ApiLayout.getInitialProps = ({ asPath }) => ({ path: asPath })
```
Comment (jescalan) [inline website/layouts/api.jsx]: any reason this changed?

Comment (jonathantneal) [reply website/layouts/api.jsx]: Yes, this was moved here from the internal `DocsPage` component. It’s a `sidenav` prop.

Comment (jescalan) [reply website/layouts/api.jsx]: I guess I mean is there any reason we don't still want to bundle it with the component, as it was previously?

**Comment by jescalan** [comment]
Ok, did a percy run on this and we're clear!

**Pull request merged**